pkg/k8s/mounterBuilder_test.go

```diff
@@ -0,0 +1,269 @@
+						Containers: []v1.Container{
+							{
+								Name:         "test-service",
+								VolumeMounts: []v1.VolumeMount{},
+							},
+						},
+					},
+				},
+			},
+		}, nil
+	})
+	k8sMock.AddReactor("get", "secrets", func(action ktesting.Action) (bool, runtime.Object, error) {
+		return true, &v1.Secret{
+			ObjectMeta: meta_v1.ObjectMeta{
+				Name: "test-secret",
+			},
+			Data: map[string][]byte{},
+		}, nil
+	})
+	k8sMock.AddReactor("Update", "deployments", func(action ktesting.Action) (bool, runtime.Object, error) {
+		return true, &v1beta1.Deployment{
+			Spec: v1beta1.DeploymentSpec{
+				Template: v1.PodTemplateSpec{
+					Spec: v1.PodSpec{
+						Volumes: []v1.Volume{},
+						Containers: []v1.Container{
+							{
+								Name:         "fh-sync-server",
+								VolumeMounts: []v1.VolumeMount{},
+							},
+						},
+					},
+				},
+			},
+		}, nil
+	})
+	return k8sMock
+}
+
+func mockMountedK8sClient() *fake.Clientset {
+	k8sMock := &fake.Clientset{}
+	k8sMock.AddReactor("list", "secrets", func(a ktesting.Action) (bool, runtime.Object, error) {
+		return true, &v1.SecretList{
+			Items: []v1.Secret{
+				{
+					Data: map[string][]byte{
+						"name": []byte("test-service"),
+					},
+				},
+				{
+					Data: map[string][]byte{
+						"now":        []byte("something"),
+						"completely": []byte("different"),
+					},
+				},
+			},
+		}, nil
+	})
+	k8sMock.AddReactor("get", "deployments", func(action ktesting.Action) (bool, runtime.Object, error) {
+		return true, &v1beta1.Deployment{
+			Spec: v1beta1.DeploymentSpec{
+				Template: v1.PodTemplateSpec{
+					Spec: v1.PodSpec{
+						Volumes: []v1.Volume{
+							{
+								Name: "test-secret",
+							},
+						},
+						Containers: []v1.Container{
+							{
+								Name: "test-service",
+								VolumeMounts: []v1.VolumeMount{
+									{
+										Name: "test-secret",
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		}, nil
+	})
+	k8sMock.AddReactor("get", "secrets", func(action ktesting.Action) (bool, runtime.Object, error) {
+		return true, &v1.Secret{
+			ObjectMeta: meta_v1.ObjectMeta{
+				Name: "test-secret",
+			},
+			Data: map[string][]byte{},
+		}, nil
+	})
+	k8sMock.AddReactor("Update", "deployments", func(action ktesting.Action) (bool, runtime.Object, error) {
+		return true, &v1beta1.Deployment{
+			Spec: v1beta1.DeploymentSpec{
+				Template: v1.PodTemplateSpec{
+					Spec: v1.PodSpec{
+						Volumes: []v1.Volume{},
+						Containers: []v1.Container{
+							{
+								Name:         "fh-sync-server",
+								VolumeMounts: []v1.VolumeMount{},
+							},
+						},
+					},
+				},
+			},
+		}, nil
+	})
+	return k8sMock
+}
+
+func TestMount(t *testing.T) {
+	cases := []struct {
+		Name      string
+		K8sClient func() *fake.Clientset
+		Namespace string
+		Service   string
+		Secret    string
+		Validate  func(t *testing.T, mountRes error)
+	}{
+		{
+			Name:      "Valid mount request",
+			K8sClient: mockUnmountedK8sClient,
```

Comment:
I think making these a little more descriptive would be good. More behaviour driven? Maybe something like 
``` should mount service secret ok```

or
``` should fail to mount when the service secret does not exist```